use-shopping-cart/src/@types/use-shopping-cart/index.d.ts

```diff
@@ -0,0 +1,105 @@
+declare module 'use-shopping-cart' {
+
+    export interface ProviderProps {
+        children: JSX.Element,
+        stripe: any,
+        successUrl: string,
+        cancelUrl: string,
+        currency: string,
+        language: string,
+        billingAddressCollection: boolean,
+        allowedCountries: null | string[]
+    }
+
+    /**
+	 * Context provider to interact with Stripe API
+	 * @param children The Entry point to your app
+	 * @param stripe The stripe object
+	 * @param successUrl The redirect url for a successful sale
+	 * @param cancelUrl The redirect url for a cancelled sale
+	 * @param currency The preferred currency
+	 * @param language The language to be used in the checkout
+	 * @param billingAddressCollection Should the billing address be collected at the checkout. Defaults to false
+     * @param allowedCountries The allowed countries
+	 */
+    export const CartProvider: (props: ProviderProps) => JSX.Element;
+
+    export interface Product {
+        /** 
+         * The name of the product
+         */
+        name?: string;
+        /**
+         * The product sku
+         */
+        sku: string;
+        /**
+         * The price of the product
+         */
+        price: number;
+        /**
+         * An image of the product
+         */
+        image?: string;
+        /**
+         * The currency of the product
+         */
+        currency?: string;
+    }
+
+    export type CartDetails = {
+        [key: string]: any
```

Comment:
`key` should probably be called `sku` and `any` should be replaced with the structure of `cartDetails` but also, it should allow any other data to be stored in each cart entry as well bc the cart entries are generated from the Products so you can still have all of your extra data like names, images, and the like (but again those aren't required pieces of data)

Reply:
ok, so let me see if i understand correctly, the cart details are basically the same as product? So could we then instead have a more generic interface that covers both use cases? Ive looked at the stipe api docs and cant find a definitive example of what it could be if it was different from product

Reply:
This is not Stripe related at all, this is entirely made up by `use-shopping-cart` and it is the data structure that is used for rendering one's cart to the screen. This definitely needs to have an actual data type rather than just `any`.

Here is an example of `cartDetails`:

```json5
{
  sku_GBJ2Ep8246qeeT: {
    name: 'Bananas',
    sku: 'sku_GBJ2Ep8246qeeT',
    price: 400,
    image: 'https://www.fillmurray.com/300/300',
    currency: 'USD',
    quantity: 3,
    formattedValue: '$12.00',
    value: 1200
  },
  sku_GBJ2WWfMaGNC2Z: {
    name: 'Tangerines',
    sku: 'sku_GBJ2WWfMaGNC2Z',
    price: 100,
    image: 'https://www.fillmurray.com/300/300',
    currency: 'USD',
    quantity: 2,
    formattedValue: '$2.00',
    value: 200
  }
}
```

It should have all of the same properties as `Product` at least but also include `formattedValue`, `value`, and `quantity`